cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -117,4 +117,9 @@ public TransactionContext newTransactionContext() throws TransactionFailureExcep
   public void dismissTransactionContext() {
     getDatasetCache().dismissTransactionContext();
   }
+
+  @Override
+  public Integer getDefaultTxTimeout() {
+    return super.getDefaultTxTimeout();
```

Comment:
This is tricky. The @Override is to implement the interface TransactionalHttpServiceContext, but the super implementation is from AbstractContext. I had to do that so that in here (https://github.com/caskdata/cdap/pull/7110/files#diff-7a3da93e5f2d71378a4a5e6d97dced82R83) I can get the defaultTimeout from the context. Casting to AbstractContext or BasicHttpServiceContext is not possible, because the HttpHandlerGeneratorTest uses a no-op implementation. I will add a comment.